File: go/testdata/fixture-multi-lang/services/jvm-suite/kotlin/PlainParser.kt
package com.example.parser

fun parseInts(s: String): List<Int> =
    s.split(",").mapNotNull { it.trim().toIntOrNull() }

fun tokenize(s: String): List<String> =
    s.split("\\s+".toRegex()).filter { it.isNotEmpty() }

data class Token(val kind: String, val value: String)

interface Parseable {
    fun parse(input: String): List<Token>
}

class SimpleParser : Parseable {
    override fun parse(input: String): List<Token> =
        tokenize(input).map { Token("word", it) }
}

File: testdata/fixture-multi-lang/services/jvm-suite/scala/PlainParser.scala
package com.example.parser

object PlainParser {
  def parseInts(s: String): List[Int] =
    s.split(",").toList.flatMap { tok =>
      tok.trim.toIntOption.toList
    }

  def tokenize(s: String): List[String] =
    s.split("\\s+").toList.filter(_.nonEmpty)
}

case class Token(kind: String, value: String)

trait Parseable {
  def parse(input: String): List[Token]
}
